data-prepper-api/src/main/java/org/opensearch/dataprepper/model/encryption/KeyProvider.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.encryption;
+
+import java.util.function.Function;
+
+public interface KeyProvider extends Function<String, byte[]> {
```

Comment:
Is there a particular reason you chose to make this a `Function`?

You could have a single method like `byte[] createKey(String)` which would be clearer.

You can also add the `@FunctionalInterface` to the class which would make support lambdas.

Reply:
Looking at how this is used, I see that this takes as input an encrypted data key. Key APIs typically work on `byte[]` over `String`. Let's change the input to `byte[]`.

How does this signature look?

```
byte[] decryptKey(byte[] encryptedKey)
```

Reply:
applied the change